packages/core/src/SpringContext.tsx

```diff
@@ -13,33 +13,48 @@ export interface SpringContext {
   immediate?: boolean
 }
 
-export const SpringContext = ({
-  children,
-  ...props
-}: PropsWithChildren<SpringContext>) => {
-  const inherited = useContext(ctx)
-
-  // Inherited values are dominant when truthy.
-  const pause = props.pause || !!inherited.pause,
-    immediate = props.immediate || !!inherited.immediate
-
-  // Memoize the context to avoid unwanted renders.
-  props = useMemoOne(() => ({ pause, immediate }), [pause, immediate])
-
-  const { Provider } = ctx
-  return <Provider value={props}>{children}</Provider>
+export const SpringContext = makeRenderableContext<
+  SpringContext,
+  PropsWithChildren<SpringContext>
+>(
+  Context =>
+    ({ children, ...props }) => {
+      const inherited = useContext(Context)
+
+      // Inherited values are dominant when truthy.
+      const pause = props.pause || !!inherited.pause
+      const immediate = props.immediate || !!inherited.immediate
+
+      // Memoize the context to avoid unwanted renders.
+      props = useMemoOne(() => ({ pause, immediate }), [pause, immediate])
+
+      return <Context value={props}>{children}</Context>
+    },
+  {} as SpringContext
+)
+
+interface RenderableContext<T, P> extends React.ProviderExoticComponent<P> {
+  Provider: RenderableContext<T, P>
+  Consumer: React.Consumer<T>
+  displayName?: string
 }
 
-const ctx = makeContext(SpringContext, {} as SpringContext)
-
-// Allow `useContext(SpringContext)` in TypeScript.
-SpringContext.Provider = ctx.Provider
-SpringContext.Consumer = ctx.Consumer
```

Comment:
This is going to be tough to make work. The component needs to have the actual context properties mutated onto it and hold a cyclic reference to itself with React 19.